v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -860,3 +851,80 @@ func ignoreNotFoundAndConflict(err error) error {
 
 	return client.IgnoreNotFound(err)
 }
+
+// ConvertResourceToARMResource converts a genruntime.MetaObject (a Kubernetes representation of a resource) into
+// a genruntime.ARMResourceSpec - a specification which can be submitted to Azure for deployment
+func (r *AzureDeploymentReconciler) ConvertResourceToARMResource(ctx context.Context) (genruntime.ARMResource, error) {
+	metaObject := r.obj
+	resolver := r.ResourceResolver
+	scheme := resolver.Scheme()
+
+	return ConvertToARMResourceImpl(ctx, metaObject, scheme, resolver, r.ARMClient.SubscriptionID())
+}
+
+// ConvertToARMResourceImpl factored out of AzureDeploymentReconciler.ConvertResourceToARMResource to allow for testing
+func ConvertToARMResourceImpl(
+	ctx context.Context,
+	metaObject genruntime.MetaObject,
+	scheme *runtime.Scheme,
+	resolver *genruntime.Resolver,
+	subscriptionID string) (genruntime.ARMResource, error) {
+	spec, err := genruntime.GetVersionedSpec(metaObject, scheme)
+	if err != nil {
+		return nil, errors.Errorf("unable to get spec from %s", metaObject.GetObjectKind().GroupVersionKind())
+	}
+
+	armTransformer, ok := spec.(genruntime.ARMTransformer)
+	if !ok {
+		return nil, errors.Errorf("spec was of type %T which doesn't implement genruntime.ArmTransformer", spec)
+	}
+
+	resourceHierarchy, resolvedDetails, err := resolve(ctx, resolver, metaObject)
+	if err != nil {
+		return nil, err
+	}
+
+	armSpec, err := armTransformer.ConvertToARM(resolvedDetails)
+	if err != nil {
+		return nil, errors.Wrapf(err, "transforming resource %s to ARM", metaObject.GetName())
+	}
+
+	typedArmSpec, ok := armSpec.(genruntime.ARMResourceSpec)
+	if !ok {
+		return nil, errors.Errorf("casting armSpec of type %T to genruntime.ARMResourceSpec", armSpec)
+	}
+
+	armID, err := resourceHierarchy.FullyQualifiedARMID(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	result := genruntime.NewARMResource(typedArmSpec, nil, armID)
+	return result, nil
+}
+
+func resolve(ctx context.Context, resolver *genruntime.Resolver, metaObject genruntime.MetaObject) (genruntime.ResourceHierarchy, genruntime.ConvertToARMResolvedDetails, error) {
```

Comment:
It might be worth keeping the TODO here - Matt was talking about reorganising stuff to make this a part of the resolver, since it's kind of a top-level driver for that.

Reply:
Done.